cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -1771,7 +1790,33 @@ Status ConvertColumnToPandas(PandasOptions options, const std::shared_ptr<Column
 
 Status ConvertTableToPandas(PandasOptions options, const std::shared_ptr<Table>& table,
                             int nthreads, MemoryPool* pool, PyObject** out) {
-  DataFrameBlockCreator helper(options, table, pool);
+  return ConvertTableToPandas(options, std::unordered_set<std::string>(), table, nthreads,
+                              pool, out);
+}
+
+Status ConvertTableToPandas(PandasOptions options,
+                            const std::unordered_set<std::string>& categorical_columns,
+                            const std::shared_ptr<Table>& table, int nthreads,
+                            MemoryPool* pool, PyObject** out) {
+  std::shared_ptr<Table> current_table = table;
+  if (categorical_columns.size() > 0) {
+    FunctionContext ctx;
+    for (int64_t i = 0; i < table->num_columns(); i++) {
+      const Column& col = *table->column(i);
+      if (categorical_columns.count(col.name())) {
+        Datum out;
+        DictionaryEncode(&ctx, Datum(col.data()), &out);
+        std::shared_ptr<ChunkedArray> array = out.chunked_array();
+        std::shared_ptr<Field> field = std::make_shared<Field>(
```

Comment:
Do we have a top level `::arrow::field` function that does this? If not, then this call should use `auto` since `Field` is already mentioned in the call to `make_shared`.